crates/polars-io/src/ipc/ipc_stream.rs

```diff
@@ -166,20 +166,14 @@ where
             self.projection = Some(prj);
         }
 
-        let sorted_projection = self.projection.clone().map(|mut proj| {
-            proj.sort_unstable();
-            proj
-        });
-
-        let schema = if let Some(projection) = &sorted_projection {
+        let schema = if let Some(projection) = &self.projection {
             apply_projection(&metadata.schema, projection)
         } else {
             metadata.schema.clone()
         };
 
-        let include_row_index = self.row_index.is_some();
         let ipc_reader =
-            read::StreamReader::new(&mut self.reader, metadata.clone(), sorted_projection);
```

Comment:
I've deleted the code that sorted the projections. I couldn't figure out why we did this and all the tests still pass when this is removed. 

A wild guess is that maybe the sorting was done to make the indexing that happens inside of `apply_projection` ordered. However it did require multiple allocations and fixing up the order later (which was broken).